### ISSUE TITLE
In-Game: [YYC] Real number gets promoted to int32, causing TestFramework to fail

### ISSUE THREAD

**Issue by DiasFranciscoA** [YoYoGames/GameMaker-Bugs#6196]
### Description

This happens when you defined a number member inside a struct that is a result of a `??` operation.

```
t = 8080
show_debug_message(typeof(t)); // real

a = {
	// when assigning a number that will be the result of a ?? operation
	b: undefined ?? 8080
}

show_debug_message(typeof(a.b)); // int32
```

DEMO:
[number_bug.zip](https://github.com/user-attachments/files/15776575/number_bug.zip)

### Expected Change

typeof shouldn't change and should be `"number"`

### Steps To Reproduce

1. Start GameMaker
2. Run attached project
3. See the issue

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.600 (Betas)

### Which platform(s) are you seeing the problem on?

Windows

### Contact Us Package Attached?

- [X] I have attached my Contact Us Package

### Sample Project Added?

- [X] I have included a small sample project

**Comment by shichen85** [comment]
This is happening to the ternary operator too, such as `condition ? a : b`

**Comment by rwkay** [comment]
Fixed in 2024.6 (once commit above has been merged)

**Comment by mgeddesGM** [comment]
verified as of  IDE v2024.600.0.577  Runtime v2024.600.0.603
